### PR TITLE
Donor Deploying

### PR DESCRIPTION
**Pull request by fs-craig**
@fs-tom, implemented donor deploy like we talked about and wrote a test to make sure this clears units from the previous demand, which was failing before the donor deploy change and passes now.  Added some helper functions to ces.rules for creating a nonbog_with_cannibals Category which uses all of the cannibalized supply and a nonbog_with_cannibals_50 Category which allows up to 50% of the cannibalized units.  Wrote a SourceFirst rule to use cannibalized units for HLD first.  Wrote some tests to make sure that's working properly as well.

### CODE REVIEW

File: src/marathon/ces/deployment.clj
Comment (fs-tom) [inline src/marathon/ces/deployment.clj:1]: check-in-demand and psuedo-deploy should be integrated better. turn c-id- into predicate, use that pred to determine is pseudo-deploy should throw or invoke pseudo-deployment.  cleaner to read.....

Comment (fs-tom) [inline src/marathon/ces/deployment.clj:1]: maybe rename donator? to donor? for consistency.

Comment (fs-tom) [inline src/marathon/ces/deployment.clj:1]: collapse donator? into valid-donor?

File: src/marathon/ces/query.clj
Comment (fs-tom) [inline src/marathon/ces/query.clj:1]: lazy :refer-all for marathon.ces.util, all the migrated functions should ideally be required by name.

File: src/marathon/ces/deployment.clj
Comment (fs-craig) [reply src/marathon/ces/deployment.clj:1]: collapsed valid-donor? into donor?

Comment (fs-craig) [reply src/marathon/ces/deployment.clj:1]: renamed donator to donor

Comment (fs-craig) [reply src/marathon/ces/deployment.clj:1]: check-in-demand is now a valid-pseudo? pred and moved exception to ces.deployment.

File: src/marathon/ces/query.clj
Comment (fs-craig) [reply src/marathon/ces/query.clj:1]: removed :refer-all and added namespace prefixes to functions in ces.query which exist  in ces.util.

**Review by fs-tom — APPROVED** [review]
Good enough.

**Pull request merged**